cmd/rpc/sock.go

```diff
@@ -0,0 +1,455 @@
+package rpc
+
+import (
+	"fmt"
+	"github.com/canopy-network/canopy/controller"
+	"github.com/canopy-network/canopy/lib"
+	"github.com/gorilla/websocket"
+	"github.com/julienschmidt/httprouter"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+)
+
+/* This file implements the client & server logic for the 'root-chain info' and corresponding 'on-demand' calls to the rpc */
+
+var _ lib.RCManagerI = new(RCManager)
+
+// RCManager handles a group of root-chain sock clients
+type RCManager struct {
+	c             lib.Config                   // the global node config
+	subscriptions map[uint64]*RCSubscription   // chainId -> subscription
+	subscribers   map[uint64]*RCSubscriber     // chainId -> subscriber
+	l             sync.Mutex                   // thread safety
```

Comment:
lock is being copied by value, that is not recommended and raise a flag in some linters (as it does on mine), please change this to a pointer reference 
```suggestion
	l             *sync.Mutex                   // thread safety
```